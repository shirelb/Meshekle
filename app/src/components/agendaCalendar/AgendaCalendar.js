import React, {Component} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {Agenda, LocaleConfig} from 'react-native-calendars';
import {localConfig} from './localConfig';
import axios from "axios";
import {SERVER_URL} from "../../shared/constants";
import phoneStorage from "react-native-simple-store";


export default class AgendaCalendar extends Component {
    constructor(props) {
        super(props);

        this.state = {
            items: {
                '2018-12-30': [{text: 'item 30 - any js object'}],
                '2018-12-31': [{text: 'item 31 - any js object'}],
                '2019-01-01': [{text: 'item 1 - any js object'}],
                '2019-01-02': [{text: 'item 1 - any js object'}],
                '2019-01-03': [{text: 'item 2 - any js object'}],
                '2019-01-04': [{text: 'item 4 - any js object'}],
                '2019-01-05': [{text: 'item 3 - any js object'}, {text: 'any js object'}],
            },
            date: new Date(),
        };

        this.userHeaders = {};
        this.userId = null;
    }

    componentDidMount() {
        phoneStorage.get('userData')
            .then(userData => {
                this.userHeaders = {
                    'Authorization': 'Bearer ' + userData.token
                };
                this.userId = userData.userId;
                this.loadItems();
            });
    }

    loadItems() {
        let newItems =
            {
                '2019-01-2': [{text: 'item 1 - any js object'}],
                '2019-01-3': [{text: 'item 2 - any js object'}],
                '2019-01-4': [],
                '2019-01-5': [{text: 'item 3 - any js object'}, {text: 'any js object'}],
            };
        axios.get(`${SERVER_URL}/api/users/events/userId/${this.userId}`,
            {headers: this.userHeaders}
        )
            .then(response => {
                console.log('response ', response);
                let events = response.data;
                // newItems
                if (events.length > 0) {
                    events.forEach((event) => {
                        switch (event.eventType) {
                            case 'Appointments':
                                axios.get(`${SERVER_URL}/api/users/appointments/userId/${this.userId}`,
                                    {
                                        headers: this.userHeaders,
                                        params: {status: 'set'}
                                    })
                                    .then(response => {
                                        console.log('get appointments response ', response);
                                        appointment
                                    })
                        }
                    })
                }
            })
            .catch(error => {
                console.log('error ', error)
            });
        this.setState({
            items: newItems
        });
    }

    onDayPress = (date) => {
        console.log('in day press ');
        this.setState({
            date: new Date(date.year, date.month - 1, date.day),
        });
    };

    onDayChange = (date) => {
        console.log('in day change ');
        this.setState({
            date: new Date(date.year, date.month - 1, date.day),
        });
    };


    renderItem(item) {
        return (
            <View style={[styles.item, {height: item.height}]}>
                <Text>{item.name}</Text>
            </View>
        );
    }

    renderEmptyDate() {
        return (
            <View style={styles.emptyDate}><Text>This is empty date!</Text></View>
        );
    }

    rowHasChanged(r1, r2) {
        return r1.name !== r2.name;
    }

    timeToString(time) {
        const date = new Date(time);
        return date.toISOString().split('T')[0];
    }

    render() {
        LocaleConfig.defaultLocale = 'il';

        let currDay = new Date; // get current date
        let currDayStr = new Date().toUTCString(); // get current date
        let first = currDay.getDate() - currDay.getDay(); // First day is the day of the month - the day of the week
        let last = first + 6; // last day is the first day + 6
        let firstDay = new Date(currDay.setDate(first)).toUTCString();
        let lastDay = new Date(currDay.setDate(last)).toUTCString();

        /*const items = {
            [format(new Date(this.state.date), 'YYYY-MM-DD')]: [],
            ...otherItems,
        };*/

        return (
            <Agenda
                items={this.state.items}
                // loadItemsForMonth={this.loadItems.bind(this)}
                selected={currDayStr}//{'2012-05-22'}
                // minDate={firstDay}//{'2012-05-20'}
                // maxDate={lastDay}//{'2012-05-27'}
                // callback that fires when the calendar is opened or closed
                // callback that gets called on day press
                // onDayPress={this.onDayPress.bind(this)}
                // callback that gets called when day changes while scrolling agenda list
                // onDayChange={this.onDayChange.bind(this)}
                renderItem={this.renderItem.bind(this)}
                renderEmptyDate={this.renderEmptyDate.bind(this)}
                rowHasChanged={this.rowHasChanged.bind(this)}
                hideKnob={true}
                // markingType={'period'}
                // markedDates={{
                //    '2017-05-08': {textColor: '#666'},
                //    '2017-05-09': {textColor: '#666'},
                //    '2017-05-14': {startingDay: true, endingDay: true, color: 'blue'},
                //    '2017-05-21': {startingDay: true, color: 'blue'},
                //    '2017-05-22': {endingDay: true, color: 'gray'},
                //    '2017-05-24': {startingDay: true, color: 'gray'},
                //    '2017-05-25': {color: 'gray'},
                //    '2017-05-26': {endingDay: true, color: 'gray'}}}
                // monthFormat={'yyyy'}
                // theme={{calendarBackground: 'red', agendaKnobColor: 'green'}}
                renderDay={(day, item) => (<Text>{day ? day.day : 'item'}</Text>)}
            />
        );
    }
}

const styles = StyleSheet.create({
    item: {
        backgroundColor: 'white',
        flex: 1,
        borderRadius: 5,
        padding: 10,
        marginRight: 10,
        marginTop: 17
    },
    emptyDate: {
        height: 15,
        flex: 1,
        paddingTop: 30
    }
});