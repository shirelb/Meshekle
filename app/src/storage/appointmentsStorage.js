import axios from "axios";
import {APP_SOCKET, SERVER_URL} from "../shared/constants";


var getUserAppointmentRequests = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointmentRequests/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'requested'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointment requests error ', error)
        });
};

var getUserAppointments = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'set'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments error ', error)
        });
};

var getUserAppointmentById = function (userId, userHeaders, eventId) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {status: 'set', appointmentId: eventId}
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments by id error ', error)
        });
};

var postUserAppointmentRequest = function (userId, serviceProvider, appointmentRequest, userHeaders) {
    return axios.post(`${SERVER_URL}/api/appointmentRequests/user/request`,
        {
            userId: userId,
            serviceProviderId: serviceProvider.serviceProviderId,
            role: serviceProvider.role,
            availableTime: appointmentRequest.availableTime,
            notes: appointmentRequest.notes ? appointmentRequest.notes : '',
            subject: JSON.stringify(appointmentRequest.subject)
        },
        {
            headers: userHeaders,
        })
        .then(response => {
            APP_SOCKET.emit('userPostAppointmentRequests', {
                serviceProviderId: serviceProvider.serviceProviderId
            });

            return response;
        })
        .catch(error => {
            console.log('post user appointment request error ', error)
        });
};

var cancelAppointmentRequestById = (appointmentRequest, headers) => {
    return axios.put(`${SERVER_URL}/api/appointmentRequests/user/reject`,
        {
            userId: appointmentRequest.AppointmentDetail.clientId,
            appointmentRequestId: appointmentRequest.requestId,
        },
        {
            headers: headers,
            params: {
                status: 'rejected'
            },
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointmentRequests', {
                serviceProviderId: appointmentRequest.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('reject appointment request error ', error);
        });
};

var cancelAppointmentById = (appointment, headers) => {
    return axios.put(`${SERVER_URL}/api/appointments/user/cancel/userId/${appointment.AppointmentDetail.clientId}/appointmentId/${appointment.appointmentId}`,
        {},
        {
            headers: headers
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointment', {
                serviceProviderId: appointment.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('cancel appointment error ', error);
        });
};


export default {
    getUserAppointments,
    getUserAppointmentById,
    postUserAppointmentRequest,
    getUserAppointmentRequests,
    cancelAppointmentRequestById,
    cancelAppointmentById,
};
