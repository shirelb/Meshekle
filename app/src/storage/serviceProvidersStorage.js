import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getServiceProviders = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('load appointments error ', error)
        });
};


export default {getServiceProviders};
