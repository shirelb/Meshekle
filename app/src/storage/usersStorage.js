import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getUserEvents = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/events/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user events error ', error)
        });
};

var getUserById = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var getUsers = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: userHeaders}
    )
        .then(response => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var userValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/users/validToken`,
        {
            "token": token,
        },
    );
}

var userLogin = function (userId, password) {
    return axios.post(`${SERVER_URL}/api/users/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
}


export default {getUserEvents, getUserById, userValidToken, userLogin, getUsers};
