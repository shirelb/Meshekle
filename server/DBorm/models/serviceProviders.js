module.exports = (sequelize, type) => {
    return sequelize.define('ServiceProviders', {
        serviceProviderId:{
            type: type.INTEGER,
            primaryKey: true,
            autoIncrement: true,
            allowNull: false,
            required: true
        },
        role : {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        userId: {
            type: type.INTEGER,
            allowNull: false,
            required: true
        },
        operationTime: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        phoneNumber: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
    });
};