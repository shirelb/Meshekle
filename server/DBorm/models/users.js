module.exports = (sequelize, type) => {
    return sequelize.define('Users', {
        userId:{
            type: type.STRING,
            primaryKey: true,
            required: true
        },
        fullname: {
            type: type.STRING,
            required: true
        },
        password: {
            type: type.STRING,
            required: true
        },
        email: {
            type: type.STRING,
            required: true
        },
        mailbox: {
            type: type.INTEGER,
            required: true
        },
        cellphone: {
            type: type.STRING,
            required: true
        },
        phone: {
            type: type.STRING
        },
        bornDate: {
            type: type.DATE,
            required: true
        },
        active: {
            type: type.BOOLEAN,
            required: true,
            defaultValue: true
        },
        image: {
            type: type.BLOB,
        }
    });
};