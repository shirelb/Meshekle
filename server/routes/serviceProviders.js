var express = require('express');
var router = express.Router();
const Sequelize = require('sequelize');
const Op = Sequelize.Op;
const {Users, ServiceProviders, AppointmentRequests, Appointments, Incidents, Events} = require('../DBorm/DBorm');

/* GET users listing. */
router.get('/', function(req, res, next) {
  res.send('respond with a resource serviceProviders');
});


/* POST user . */
router.post('/users/add', function (req, res, next) {
    Users.create({
        userId: req.body.userId,
        fullname: req.body.fullname,
        password: req.body.password,
        email: req.body.email,
        mailbox: req.body.mailbox,
        cellphone: req.body.cellphone,
        phone: req.body.phone,
        bornDate: new Date(req.body.bornDate),
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/* POST user . */
router.post('/add', function (req, res, next) {
    ServiceProviders.create({
        serviceProviderId: req.body.serviceProviderId,
        userId: req.body.userId,
        role: req.body.role,
        operationTime: req.body.operationTime,
        phoneNumber: req.body.phoneNumber,
        appointmentWayType: req.body.appointmentWayType,
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

module.exports = router;
