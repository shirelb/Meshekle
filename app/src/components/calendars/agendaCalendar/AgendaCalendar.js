import React, {Component} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {Avatar, Card} from 'react-native-elements';
import {Agenda, LocaleConfig} from 'react-native-calendars';
import '../localConfig';
import phoneStorage from "react-native-simple-store";
import usersStorage from "../../../storage/usersStorage";
import moment from "moment";
import 'moment/locale/he.js';
import mappers from "../../../shared/mappers";

moment.locale('he');

const module2color = {
    Appointments: 'purple', //'#00adf5',
    UsersChores: 'red',
    Announcements: 'green',
};

const module2selectedDotColor = {
    Appointments: 'purple',
    UsersChores: 'red',
    Announcements: 'green',
};


export default class AgendaCalendar extends Component {
    constructor(props) {
        super(props);

        this.state = {
            refreshing: false,
            items: {}
            /* '2018-12-30': [{text: 'item 30 - any js object'}],
             '2018-12-31': [{text: 'item 31 - any js object'}],
             '2019-01-01': [{text: 'item 1 - any js object'}],
             '2019-01-02': [{text: 'item 1 - any js object'}],
             '2019-01-03': [{text: 'item 2 - any js object'}],
             '2019-01-04': [{text: 'item 4 - any js object'}],
             '2019-01-05': [{text: 'item 3 - any js object'}, {text: 'any js object'}],
         },*/
        };

        this.userHeaders = {};
        this.userId = null;
    }

    componentDidMount() {
        phoneStorage.get('userData')
            .then(userData => {
                console.log('agenda componentDidMount userData ', userData);
                this.userHeaders = {
                    'Authorization': 'Bearer ' + userData.token
                };
                this.userId = userData.userId;

                this.serviceProviders = [];
                usersStorage.getUsers(this.userHeaders)
                    .then(serviceProvidersFound => {
                        this.serviceProviders = serviceProvidersFound.filter(user => user['ServiceProviders'].length > 0);

                        this.loadItems();
                    });
            });
    }

    onRefresh = () => {
        this.setState({refreshing: true});

        this.loadItems();
    };

    loadItems() {
        let newItems = {};
        let serviceProviderUserDetails = this.serviceProviders;
        usersStorage.getUserEvents(this.userId, this.userHeaders)
            .then(response => {
                let events = response.data;
                if (events.length > 0) {
                    events.forEach((event) => {
                        let item = {};
                        switch (event.eventType) {
                            case 'Appointments':
                                let appointment = event['ScheduledAppointment'];
                                item.type = event.eventType;
                                item.itemId = appointment.appointmentId;
                                item.date = moment(appointment.startDateAndTime).format("YYYY-MM-DD");
                                item.startTime = moment(appointment.startDateAndTime).format("HH:mm");
                                item.endTime = moment(appointment.endDateAndTime).format("HH:mm");
                                item.role = mappers.serviceProviderRolesMapper(appointment.AppointmentDetail.role);
                                item.serviceProviderId = appointment.AppointmentDetail.serviceProviderId;
                                item.subject = JSON.parse(appointment.AppointmentDetail.subject).join(", ");
                                let serviceProvider = serviceProviderUserDetails.filter(provider => provider.userId === appointment.AppointmentDetail.serviceProviderId.toString())[0];
                                item.serviceProviderFullname = serviceProvider.fullname;
                                item.serviceProviderImage = serviceProvider.image;
                                if (newItems[item.date]) {
                                    newItems[item.date].push(item);
                                } else {
                                    newItems[item.date] = [item];
                                }
                                break;
                            case 'UsersChores':
                                console.log("load items-> userschores", this.state.items)
                                let chore = event['UsersChore'];
                                item.type = event.eventType;
                                item.date = moment(chore.date).format("YYYY-MM-DD");
                                ///item.itemId = chore.userChoreId;
                                ////item.date = moment(chore.date).format("YYYY-MM-DD");
                                item.title = chore.choreTypeName;
                                item.startTime = moment(chore.date).format("HH:mm");
                                item.endTime = moment(chore.date).format("HH:mm");
                                //item.role = mappers.serviceProviderRolesMapper(appointment.AppointmentDetail.role);
                                //item.serviceProviderId = appointment.AppointmentDetail.serviceProviderId;
                                //item.subject = JSON.parse(appointment.AppointmentDetail.subject).join(", ");
                                //let serviceProvider = serviceProviderUserDetails.filter(provider => provider.userId === appointment.AppointmentDetail.serviceProviderId.toString())[0];
                                //item.serviceProviderFullname = serviceProvider.fullname;
                                //item.serviceProviderImage = serviceProvider.image;
                                if (newItems[item.date]) {
                                    newItems[item.date].push(item);
                                } else {
                                    newItems[item.date] = [item];
                                }
                                break;
                            case 'Announcements': {
                                let announcement = event['Announcement'];
                                item.type = event.eventType;
                                item.itemId = announcement.announcementId;
                                item.date = moment(announcement.dateOfEvent).format("YYYY-MM-DD");
                                item.title = announcement.title;
                                item.content = announcement.content;

                                if (newItems[item.date]) {
                                    newItems[item.date].push(item);
                                } else {
                                    newItems[item.date] = [item];
                                }
                                break;
                            }
                        }
                    });
                    this.setState({
                        items: newItems,
                        refreshing: false
                    });
                }
            })
    }

    onDayPress = (date) => {
        console.log('in day press ');
        this.setState({
            date: moment(date).subtract(1, 'months'),
        });
    };


    renderItem = (item) => {
        switch (item.type) {
            case 'Appointments':
                return (
                    <Card
                        title={`תור ל ${item.role} - ${item.serviceProviderFullname}`}
                        // containerStyle={{width: 70 + '%'}}
                    >
                        <View style={{
                            flex: 1,
                            flexDirection: 'row',
                            justifyContent: 'space-between',
                        }}>
                            <Text style={{marginBottom: 10}}>{item.startTime} - {item.endTime}</Text>

                            <Avatar
                                width={50}
                                rounded
                                source={{uri: item.serviceProviderImage}}
                                // avatarStyle={{height: 40, width: 40}}
                            />
                        </View>
                        <Text h3>{item.subject}</Text>
                    </Card>
                );
            case 'UsersChores':
                return (
                    <Card
                        title={`תורנות`}
                        // containerStyle={{width: 70 + '%'}}
                    >
                        <View style={{
                            flex: 1,
                            flexDirection: 'row',
                            justifyContent: 'space-between',
                        }}>
                            <Text style={{marginBottom: 10}}>{item.title}</Text>


                        </View>
                    </Card>
                );
            case 'Announcements': {
                return (
                    <Card
                        title={`מודעה - ${item.title} `}
                        // containerStyle={{width: 70 + '%'}}
                    >
                        <View style={{
                            flex: 1,
                            flexDirection: 'row',
                            justifyContent: 'space-between',
                        }}>
                            <Text style={{marginBottom: 10}}>{item.dateOfEvent}</Text>


                        </View>
                        <Text h3>{item.content}</Text>
                    </Card>
                );
            }

        }
    };

    renderDay = (day) => {
        if (day !== undefined) {
            let color = moment().format("YYYY-MM-DD") === day.dateString ? '#00adf5' : '#2d4150';
            return (
                <View style={{marginTop: 15, marginLeft: 15}}>
                    <Text style={[styles.day, {color: color}]}>{day.day} </Text>
                    <Text
                        style={[styles.dayMonth, {color: color}]}>{moment(day.dateString).format("MMM")} </Text>
                    <Text
                        style={[styles.dayMonth, {color: color}]}>{moment(day.dateString).format("dddd")} </Text>
                </View>
            );
        } else {
            return <View style={{width: 18 + '%'}}><Text> </Text></View>
        }
    };

    renderEmptyDate = () => {
        // return <View style={{width: 18 + '%'}}><Text> אין אירועים בתאריך זה </Text></View>
        return <View style={styles.emptyDate}><Text> אין אירועים בתאריך זה </Text></View>
    };

    rowHasChanged = (r1, r2) => {
        return r1.itemId !== r2.itemId;
    };


    render() {
        console.log("agendacalendar state ", this.state);

        LocaleConfig.defaultLocale = 'il';

        // let currDay = new Date; // get current date
        // let currDayStr = new Date().toUTCString(); // get current date
        // let first = currDay.getDate() - currDay.getDay(); // First day is the day of the month - the day of the week
        // let last = first + 6; // last day is the first day + 6
        // let firstDay = new Date(currDay.setDate(first)).toUTCString();
        // let lastDay = new Date(currDay.setDate(last)).toUTCString();

        const today = moment();
        const from_date = today.startOf('week');
        const to_date = today.endOf('week');

        return (
            <Agenda
                items={this.state.items}
                // loadItemsForMonth={this.loadItems.bind(this)}
                // selected={today.format("YYYY-MM-DD")}//{'2012-05-22'}
                // minDate={from_date}//{'2012-05-20'}
                // maxDate={to_date}//{'2012-05-27'}
                // callback that fires when the calendar is opened or closed
                // callback that gets called on day press
                // onDayPress={this.onDayPress}
                // callback that gets called when day changes while scrolling agenda list
                // onDayChange={this.onDayChange}
                renderItem={this.renderItem}
                renderEmptyDate={this.renderEmptyDate}
                renderEmptyData={this.renderEmptyDate}
                renderDay={this.renderDay}
                rowHasChanged={this.rowHasChanged}
                onRefresh={this.onRefresh}
                refreshing={this.state.refreshing}
                refreshControl = {null}
                // hideKnob={true}
                // markingType={'period'}
                // markedDates={{
                //     '2019-01-08': {textColor: '#666'},
                //     '2019-01-09': {textColor: '#666'},
                //     '2019-01-04': {startingDay: true, endingDay: true, color: 'blue'},
                //     '2019-01-01': {startingDay: true, color: 'blue'},
                //     '2019-01-02': {endingDay: true, color: 'gray'},
                //     '2019-01-03': {startingDay: true, color: 'gray'},
                //     '2019-01-05': {color: 'gray'},
                //     '2019-01-06': {endingDay: true, color: 'gray'}
                // }}
                // monthFormat={'yyyy'}
                // theme={{calendarBackground: 'red', agendaKnobColor: 'green'}}
                // Specify theme properties to override specific styles for calendar parts. Default = {}
                theme={{
                    backgroundColor: '#f7f7f7',
                    calendarBackground: '#ffffff',
                    textSectionTitleColor: '#5a646d',
                    selectedDayBackgroundColor: '#00adf5',
                    selectedDayTextColor: '#ffffff',
                    todayTextColor: '#00adf5',
                    dayTextColor: '#2d4150',
                    textDisabledColor: '#d9e1e8',
                    dotColor: '#00adf5',
                    selectedDotColor: '#ffffff',
                    arrowColor: 'orange',
                    monthTextColor: 'blue',
                    textDayFontFamily: 'monospace',
                    textMonthFontFamily: 'monospace',
                    textDayHeaderFontFamily: 'monospace',
                    textMonthFontWeight: 'bold',
                    textDayFontSize: 16,
                    textMonthFontSize: 16,
                    textDayHeaderFontSize: 16,
                    agendaDayTextColor: 'yellow',
                    agendaDayNumColor: 'green',
                    agendaTodayColor: 'orange',
                    agendaKnobColor: 'blue',
                    agendaBackgroundColor: '#424242',
                }}
                style={styles.calendar}
            />
        );
    }
}

const styles = StyleSheet.create({
    calendar: {
        // borderTopWidth: 2,
        // marginTop: 5,
        paddingTop: 5,
        // borderBottomWidth: 2,
        // borderColor: '#eee',
        height: 500,
    },
    item: {
        backgroundColor: 'white',
        flex: 1,
        borderRadius: 5,
        padding: 10,
        marginRight: 10,
        marginTop: 17
    },
    emptyDate: {
        flex: 1, justifyContent: "center", alignItems: "center",
        // height: 15,
        // flex: 1,
        // paddingTop: 30,
        borderTopWidth: 2,
        borderTopColor: 'grey',
        borderBottomWidth: 2,
        borderBottomColor: 'grey',
        // textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 18,
    },
    dayMonthContainer: {
        height: 100,
        // borderTopWidth: 2,
        // borderTopColor: 'grey',
        // marginRight: 30
    },
    day: {
        fontSize: 20,
        fontWeight: '300',
        // color: appStyle.agendaDayMonthColor,
        marginTop: -5,
        backgroundColor: 'rgba(0,0,0,0)'
    },
    dayMonth: {
        fontSize: 14,
        fontWeight: '300',
        // color: appStyle.agendaDayMonthColor,
        marginTop: -5,
        backgroundColor: 'rgba(0,0,0,0)'
    },
});