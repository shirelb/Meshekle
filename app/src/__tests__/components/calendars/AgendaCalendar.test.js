import React from 'react';

import {Card} from 'react-native-elements';
import {Agenda} from 'react-native-calendars';
import {shallow} from "enzyme/build";
import AgendaCalendar from "../../../components/calendars/agendaCalendar/AgendaCalendar";
import usersStorage from "../../../storage/usersStorage";
import phoneStorage from "react-native-simple-store";
import users from "../../jsons/users";
import serviceProvidersStorage from "../../../storage/serviceProvidersStorage";
import serviceProviders from "../../jsons/serviceProviders";
import appointmentsStorage from "../../../storage/appointmentsStorage";
import user013637605AppointmentRequests from "../../jsons/user013637605AppointmentRequests";
import user135985058Events from "../../jsons/user135985058Events";

jest.mock("../../../storage/serviceProvidersStorage");
jest.mock("../../../storage/usersStorage");
jest.mock("react-native-simple-store");


describe("AgendaCalendar should", () => {
    let wrapper = null;
    let componentInstance = null;
    const userIdTest = "135985058";
    const props = {};
    const mockStore = {
        userData: {
            userId: userIdTest,
            token: "some token"
        }
    };

    phoneStorage.get = jest.fn().mockImplementation((key) => Promise.resolve(mockStore[key]));
    phoneStorage.update = jest.fn().mockImplementation((key, value) => Promise.resolve(mockStore[key] = value));
    usersStorage.getUsers = jest.fn().mockImplementation(() => Promise.resolve(users));
    usersStorage.getUserEvents = jest.fn().mockImplementation(() => Promise.resolve({data: user135985058Events}));
    usersStorage.getUserByUserID = jest.fn().mockImplementation((userId) => Promise.resolve(users.filter(user => user.userId === userId)[0]));
    serviceProvidersStorage.getServiceProviders = jest.fn().mockResolvedValue(serviceProviders);
    serviceProvidersStorage.getServiceProviderUserDetails = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve({data: users.filter(user => user.userId === serviceProviderId)[0]}));
    serviceProvidersStorage.getServiceProviderById = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve(serviceProviders.filter(provider =>
        provider.serviceProviderId === serviceProviderId)));
    appointmentsStorage.getUserAppointmentRequests = jest.fn().mockImplementation(() => Promise.resolve(user013637605AppointmentRequests));
    appointmentsStorage.postUserAppointmentRequest = jest.fn().mockImplementation(() => Promise.resolve());
    appointmentsStorage.cancelAppointmentById = jest.fn().mockImplementation(() => Promise.resolve());

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('match snapshot', async () => {
        wrapper = shallow(<AgendaCalendar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("render what the user see", async () => {
        wrapper = await shallow(<AgendaCalendar {...props} />);
        componentInstance = wrapper.instance();

        await wrapper.update();

        expect(wrapper.find(Agenda)).toHaveLength(1);
        expect(wrapper.find(Card)).toHaveLength(0);
    });

    it("mounted with the right data", async () => {
        expect(Object.keys(componentInstance.state.items).length).toEqual(6);
        expect(Object.values(componentInstance.state.items)[0][0].type).toEqual('Announcements');
        expect(Object.values(componentInstance.state.items)[1][0].type).toEqual('Announcements');
        expect(Object.values(componentInstance.state.items)[2][0].type).toEqual('Appointments');
        expect(Object.values(componentInstance.state.items)[3][0].type).toEqual('Appointments');
        expect(Object.values(componentInstance.state.items)[4][0].type).toEqual('Appointments');
    });

});
