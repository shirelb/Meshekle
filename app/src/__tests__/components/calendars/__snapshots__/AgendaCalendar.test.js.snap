// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgendaCalendar should match snapshot 1`] = `
<AgendaView
  items={
    Object {
      "2019-06-04": Array [],
    }
  }
  renderDay={[Function]}
  renderEmptyDate={[Function]}
  renderItem={[Function]}
  rowHasChanged={[Function]}
  theme={
    Object {
      "agendaBackgroundColor": "#424242",
      "agendaDayNumColor": "green",
      "agendaDayTextColor": "yellow",
      "agendaKnobColor": "blue",
      "agendaTodayColor": "orange",
      "arrowColor": "orange",
      "backgroundColor": "#f7f7f7",
      "calendarBackground": "#ffffff",
      "dayTextColor": "#2d4150",
      "dotColor": "#00adf5",
      "monthTextColor": "blue",
      "selectedDayBackgroundColor": "#00adf5",
      "selectedDayTextColor": "#ffffff",
      "selectedDotColor": "#ffffff",
      "textDayFontFamily": "monospace",
      "textDayFontSize": 16,
      "textDayHeaderFontFamily": "monospace",
      "textDayHeaderFontSize": 16,
      "textDisabledColor": "#d9e1e8",
      "textMonthFontFamily": "monospace",
      "textMonthFontSize": 16,
      "textMonthFontWeight": "bold",
      "textSectionTitleColor": "#5a646d",
      "todayTextColor": "#00adf5",
    }
  }
/>
`;
